subsys/usb/device/class/cdc_acm.c

```diff
@@ -57,6 +57,7 @@
 
 #include <zephyr/logging/log.h>
 #if DT_NODE_HAS_COMPAT(DT_CHOSEN(zephyr_console), zephyr_cdc_acm_uart) \
+	&& defined(CONFIG_UART_CONSOLE)	\
```

Comment:
If it is about backends why not
```c
	&& defined(CONFIG_LOG_BACKEND_UART) \
	&& defined(CONFIG_SHELL_BACKEND_SERIAL) \
```
?

Reply:
It seems that we could use `CONFIG_LOG_BACKEND_UART` here (when logging is disabled, there is actually no need to update the log level here).

Reply:
Shell use-case would need a separate check here. It relies on `zephyr, shell-uart` DT chosen and only forwards log if `CONFIG_SHELL_LOG_BACKEND` is enabled. I would suggest to add check for shell use-case in a separate PR then.

Reply:
The PR could also take into account `zephyr,log-uart` DT chosen which can also be used by UART log backend.

Reply:
> Shell use-case would need a separate check here. It relies on `zephyr, shell-uart` DT chosen and only forwards log if `CONFIG_SHELL_LOG_BACKEND` is enabled. I would suggest to add check for shell use-case in a separate PR then.

Please do it in this PR.